app/src/main/java/com/example/weatherandroidapp/utils/WeatherApp.kt

```diff
@@ -0,0 +1,24 @@
+package com.example.weatherandroidapp.utils
+
+import android.app.Application
+import android.content.Context
+import com.example.weatherandroidapp.di.viewModelsModule
+import org.koin.android.ext.koin.androidContext
+import org.koin.android.ext.koin.androidLogger
+import org.koin.core.context.startKoin
+import org.koin.core.logger.Level
+
+class WeatherApp : Application() {
+    override fun onCreate() {
+        super.onCreate()
+        startKoin {
+            androidContext(this@WeatherApp)
+            androidLogger(Level.DEBUG)
+            modules(listOf(viewModelsModule))
+        }
+    }
+
+    override fun attachBaseContext(base: Context?) {
```

Comment:
as you are only calling the super, this is unnecessary, please remove this method

Reply:
Done